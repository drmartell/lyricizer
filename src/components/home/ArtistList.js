import React from 'react';
import { Link } from 'react-router-dom';
import ArtistItem from './ArtistsItem';
import PropTypes from 'prop-types';

const ArtistList = ({ artistArr }) => {
  const artistsElements = artistArr.map(artist =>{
    const kabobArtist = artist.name.replace(' ', '-').toLowerCase();
    return <li key={artist.id}>
      <Link to={`/artist/${artist.id}?artist=${kabobArtist}`}>
        <ArtistItem name={artist.name} />
      </Link>
    </li>;
  });

  return (
    <ul>
      {artistsElements}
    </ul>
  );
};

ArtistList.propTypes = {
  artistArr: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired
  })).isRequired
};

export default ArtistList;
