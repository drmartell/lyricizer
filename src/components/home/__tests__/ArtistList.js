import React from 'react';
import { shallow } from 'enzyme';
import ArtistList from '../ArtistList';

describe('ArtistList Component', () => {
  it('should render Artists List', () => {
    const wrapper = shallow(<ArtistList artistArr={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
