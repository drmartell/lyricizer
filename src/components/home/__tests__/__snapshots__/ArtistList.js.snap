// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistList Component should render Artists List 1`] = `<ul />`;
